sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/hash.scala

```diff
@@ -296,7 +297,7 @@ abstract class HashExpression[E] extends Expression {
 
     ev.copy(code =
       code"""
-         |$hashResultType ${ev.value} = $seed;
+         |$hashResultType ${ev.value} = $typedSeed;
```

Comment:
@patrickcording  can we add a simple test?

Reply:
Sure, I added this https://github.com/apache/spark/pull/27354/commits/77bfb37d510fd9126e6878fc989391254c1dbb72